marconi-chain-index/src/Marconi/ChainIndex/Types.hs

```diff
@@ -56,6 +60,9 @@ txOutRef = C.TxIn
 newtype SecurityParam = SecurityParam Word64
   deriving newtype (Eq, Ord, Bounded, Enum, Real, Num, Read, Integral, Show)
 
+newtype TxIndexInBlock = TxIndexInBlock Word64
+  deriving newtype (Eq, Ord, Bounded, Enum, Real, Num, Read, Integral, Show, Aeson.FromJSON, Aeson.ToJSON, SQL.ToField, SQL.FromField)
```

Comment:
not clear why we need to derive `Real`

Reply:
@kayvank Real is required for Integral, no other reason. But why is it a super-class of Integral I've no idea -- what I remember from math class is that it should be the other way around (Natural -> Integral -> Rational -> Real).

Reply:
I've decided to stop trying to understand and/or memorise the Haskell numeric type classes hierarchy and content. I lookup at the content of the typeclasses when I need something and that's it. I know there was a few tries to set up alternatives, but for compatibility reasons, they failed / are failing.